File: CSharp/test/test_tabulate/test_tabulate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThrustRTCSharp;

namespace test_tabulate
{
    class test_tabulate
    {
        static void Main(string[] args)
        {
            DVVector vec = new DVVector("int32_t", 10);
            TRTC.Sequence(vec);
            TRTC.Tabulate(vec, new Functor("Negate"));
            print_array((int[])vec.to_host());
        }

        static void print_array<T>(T[] arr)
        {
            foreach (var item in arr)
            {
                Console.Write(item.ToString());
                Console.Write(" ");
            }
            Console.WriteLine("");
        }
    }
}

File: java/test/test_tabulate.java
import java.util.*;
import JThrustRTC.*;

public class test_tabulate
{
	public static void main(String[] args) 
	{	
        DVVector vec = new DVVector("int32_t", 10);
        TRTC.Sequence(vec);
        TRTC.Tabulate(vec, new Functor("Negate"));
        System.out.println(Arrays.toString((int[])vec.to_host()));
	}	

}
